### ISSUE TITLE
Possibly confusing wording in Step 5?

### ISSUE THREAD

**Issue by epylar** [kanaka/mal#453]
The guide's step 5 says, for the default 'apply' case --

The default "apply"/invoke case of EVAL must now be changed to account for the new object/structure returned by the fn* form. Continue to call eval_ast on ast. The first element is f. 

Switch on the type of f:

* regular function (not one defined by fn*): apply/invoke it as before (in step 4).

* a fn* value: set ast to the ast attribute of f. Generate a new environment using the env and params attributes of f as the outer and binds arguments and rest ast arguments (list elements 2 through the end) as the exprs argument. Set env to the new environment. Continue at the beginning of the loop.

.....

for "rest ast arguments" I have tried --
rest of: the ast attribute of f
rest of: original ast before eval
rest of: evaluated ast

And only the third works (which seems to make sense--you evaluate the original arguments to the function and then pass them in).  This took me a while to figure out because I was trying to stick to the guide as closely as possible at first.

**Comment by kanaka** [comment]
Yes you're right, that is certainly an oversight. Would you like to submit a PR with the fix?

Out of curiosity, what language are you working on?

**Comment by epylar** [comment]
@kanaka Sure, I can probably get one in this weekend. I'm doing Python 3 with type annotations and using the Arpeggio parser for Python.. the former to get some practice with mypy and the latter just for fun.

**Comment by epylar** [comment]
Some of the core grammar --

```# Arpeggio grammar
def mExpression():
    return [mArray, mList, mInt, mString, mNil, mBoolean, mSymbol]


def mList():
    return "(", ZeroOrMore(mExpression), ")"```

**Comment by kanaka** [comment]
Cool. Sounds like it might be unique enough from the existing python to be merged as a separate python implementation (see ada and ada.2)

**Comment by kanaka** [comment]
Resolved by #454
